src/org/zalando/stups/magnificent/api.clj

```diff
@@ -187,26 +187,15 @@
               (log/info "Access rejected: %s" {:reason "Not a team member and team has no accounts" :team team :member-id member-id})
               (api/throw-error 403 "Not a team member." {:member-id member-id :team team}))
             ; start fetching accounts async
-            (doseq [account accounts]
-              (go
-                (put! channel (account/get-account
-                                account-api
-                                (:type account)
-                                (:id account)
-                                token))))
-            ; wait for them sync one by one
-            (loop [account-data (<!! channel)
-                   counter      1]
-              (when-not (util/account-member? account-data member-id)
-                ; throw when we're at the last account already and user is not a member
-                (when (and (= counter nr-of-accounts))
-                  (log/info "Access rejected: %s" {:reason "No access to any team account" :team team :member-id member-id})
-                  (api/throw-error 403 "Not a team member or no access to its accounts." {:member-id member-id :team team}))
-                ; recur when there is at least one more account to look at
-                (when (< counter nr-of-accounts)
```

Comment:
It's looping through all accounts, if any account failed to pass ```account-member?```, we reject the access.
So all accounts data must be fetched anyway.
I assume ```get-account``` is blocking. It's also bad to block in go block :)